Tensile/Source/lib/include/Tensile/ConvolutionProblem.hpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * MIT License
+ *
+ * Copyright (C) 2019 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell cop-
+ * ies of the Software, and to permit persons to whom the Software is furnished
+ * to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IM-
+ * PLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNE-
+ * CTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#pragma once
+
+#include <Tensile/Tensile.hpp>
+#include <Tensile/ContractionProblem_fwd.hpp>
+#include <vector>
+
+namespace Tensile
+{
+
+    class TENSILE_API ConvolutionProblem
+    {
+    public:
+        static const size_t MaxNumSpatialDims=3;
+        static const size_t InvalidPos;
+        enum class TensorFormat
+        {
+            NCHW, NHWC, CNHW, // Activation Formats
+            KCYX, CKYX        // Weight Formats
+        };
+
+        class TENSILE_API BaseFormat {
+        public:
+            std::string formatIdentifier() const { return m_formatIdentifier; };
+            TensorFormat format() const { return m_format;};
+        protected:
+            std::string  m_formatIdentifier;
+            TensorFormat m_format;
+        };
+
+        class TENSILE_API ActivationFormat : public BaseFormat {
+        public:
+            ActivationFormat();
+            void FromIdentifier(std::string identifier, size_t formatNumSpatialDims,
+                    size_t numSpatialDims, std::vector<size_t> *filters);
+            std::string description() const;
+
+            size_t batchPosition() const { return m_batchPosition; };
+            size_t channelPosition() const   { return m_channelPosition; };
+
+            // fi is filter index 0...MaxNumSpatialDims.
```

Comment:
- When we have these explanatory comments, they should be Doxygen style comments so we can eventually do something with them.
 - What is 'fi' in this context? I don't see that anywhere around here.